client/add/add.ctrl.js

```diff
@@ -123,11 +128,16 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
-      .tap(box.onscroll)
+      .filter(line => boxCounts[line.box]++ < BOX_PAGE_SIZE)
```

Comment:
This isn't doing the same thing as the previous code. Previously, if a box had 45 pokemon and a bunch of new uploads were added, the amount would jump to 60 (filling the current box page). Now it will stay at 45 (because it's larger than a single box page).

I think using something along the lines of the `length % BOX_PAGE_SIZE` logic from before would be better.

(Also, I'm not really a big fan of having a filter function with side-effects, but it's not a PR-blocker.)